addons/point_of_sale/models/point_of_sale_order.py

```diff
@@ -0,0 +1,723 @@
+            'payment_date': ui_paymentline['name'],
+            'statement_id': ui_paymentline['statement_id'],
+            'payment_name': ui_paymentline.get('note', False),
+            'journal':      ui_paymentline['journal_id'],
+        }
+
+    def _get_rescue_session(self, order):
+        """ Find or generate a rescue session """
+        Session = self.env['pos.session']
+        date = order.get('creation_date', time.strftime(tools.DEFAULT_SERVER_DATETIME_FORMAT))
+        closed_session = Session.browse(order['pos_session_id'])
+        rescue_session = Session.search([
+            ('rescue', '=', True), ('config_id', '=', closed_session.config_id.id),
+            ('start_at', '<=', date), ('state', '=', 'opened')
+        ], limit=1)
+        if not rescue_session:
+            return closed_session.copy(default={'rescue': True})
+        return rescue_session[0]
+
+    @api.model
+    def _process_order(self, order):
+        pos_session = self.env['pos.session'].browse(order['pos_session_id'])
+        if pos_session.state == 'closed':
+            rescue_session_id = self._get_rescue_session(order)
+            order['pos_session_id'] = rescue_session_id
+        pos_order = self.create(self._order_fields(order))
+
+        for payments in order['statement_ids']:
+            self.add_payment(pos_order.id, self._payment_fields(payments[2]))
+
+        if pos_session.sequence_number <= order['sequence_number']:
+            pos_session.write({'sequence_number': order['sequence_number'] + 1})
+            pos_session.refresh()
+
+        if not float_is_zero(order['amount_return'], self.env['decimal.precision'].precision_get('Account')):
+            cash_journal = pos_session.cash_journal_id
+            if not cash_journal:
+                cash_journal_ids = filter(lambda st: st.journal_id.type == 'cash', pos_session.statement_ids)
+                if not len(cash_journal_ids):
+                    raise UserError(_("No cash statement found for this session. Unable to record returned cash."))
+                cash_journal = cash_journal_ids[0].journal_id
+            self.add_payment(pos_order.id, {
+                'amount': -order['amount_return'],
+                'payment_date': time.strftime('%Y-%m-%d %H:%M:%S'),
+                'payment_name': _('return'),
+                'journal': cash_journal.id,
+            })
+        return pos_order
+
+    def _prepare_analytic_account(self, line):
+        '''This method is designed to be inherited in a custom module'''
+        return False
+
+    def _create_account_move(self, dt, ref, journal_id, company_id):
+        start_at_datetime = datetime.strptime(dt, tools.DEFAULT_SERVER_DATETIME_FORMAT)
+        date_tz_user = fields.Datetime.context_timestamp(self, start_at_datetime)
+        date_tz_user = date_tz_user.strftime(tools.DEFAULT_SERVER_DATE_FORMAT)
+        return self.env['account.move'].sudo().create({'ref': ref, 'journal_id': journal_id, 'date': date_tz_user})
```

Comment:
Sudo is required here, otherwise pos user can not allowed to do a validate closing& post entries.. :)